src/secret.rs

```diff
@@ -113,33 +114,31 @@ impl SecretKey {
     /// little-endian hexadecimal string using the provided formatter.
     ///
     /// This is the only method that outputs the actual secret key value, and, thus,
-    /// should be used with extreme precaution.
+    /// should be used with extreme caution.
     ///
-    /// # Example
+    /// # Examples
     ///
     /// ```
     /// # #[cfg(all(feature = "std", not(feature = "bitcoin_hashes")))] {
-    /// use secp256k1::ONE_KEY;
-    /// let key = ONE_KEY;
-    /// // Normal display hides value
-    /// assert_eq!(
-    ///     "SecretKey(#2518682f7819fb2d)",
-    ///     format!("{:?}", key)
-    /// );
+    /// let key = secp256k1::ONE_KEY;
+    ///
+    /// // Normal display hides value.
+    /// assert_eq!("SecretKey(#2518682f7819fb2d)", format!("{:?}", key));
```

Comment:
Oh, I see now, this is actually hard coded. The test is wrong because `std` doesn't guarantee any particular hash function or value. I suggest to test it with `bitcoin_hashes` only.

Reply:
I think `std` should guarantee that the shared secrets don't change, and it is a breaking change if they do.

Having said this, we shouldn't guarantee the *debug output* of anything.

Reply:
They don't guarantee it in case a better hash function is available so it can be switched. Notice that it's specifically called `DefaultHasher` - there was `SipHasher` before. I believe that's fine, if people want a specific hash function they should just explicitly use it (as a crate or NIH).

Reply:
So I wonder how this passed CI when I originally implemented it (and I think, IIRC, it was like this before I touched it also)?

Reply:
Happened to be the same hasher version that produced the expected output. So different versions of Rust produce different hashes - as documented.

Reply:
Left an example of debug output in the rustdocs of `SecretKey`, removed the assertion. Removed all mention of debug output for the other two secrets (`KeyPair` and `SharedSecret`).

Reply:
> They don't guarantee it in case a better hash function is available so it can be switched. Notice that it's specifically called `DefaultHasher` - there was `SipHasher` before. I believe that's fine, if people want a specific hash function they should just explicitly use it (as a crate or NIH).

I really don't like this. The implication that if users try to share a secret, and one user has a slightly different version of rust-secp than the other, that they will be unable to communicate because their crypto doesn't work.

Reply:
That's non-issue because we only use this for debug output which must not be relied on. Creation of `SharedSecret` still uses sha256.